stdlib/public/SDK/Foundation/Array+DataProtocol.swift

```diff
@@ -0,0 +1,65 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+
+extension ArraySlice: DataProtocol where Element == UInt8 {
+    public var regions: CollectionOfOne<ArraySlice<UInt8>> {
+        return CollectionOfOne(self)
+    }
+    
+    public var _borrowedBytes: (buffer: UnsafeRawBufferPointer, owner: Any)? {
+        return withUnsafeBytes { (buffer) -> (buffer: UnsafeRawBufferPointer, owner: Any) in
```

Comment:
This, of course, pending approval from the stdlib for pulling tricks like this. The alternative is to move `_borrowedBytes` down into `Sequence` or something, which would really give us what we want: a way to get partial ownership of the underlying buffer without copying.